From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Richard Townsend <richard.townsend@arm.com>
Date: Wed, 24 Apr 2019 13:57:36 +0100
Subject: Workaround an undefined symbol error

Previously, electron builds configured with dcheck_always_on=true would
error out with messages like this in the log:

 lld-link: error: undefined symbol: public: bool __cdecl
   v8::internal::CPURegister::IsZero(void) const

This problem is caused by electron's unusual build configuration, where
symbols are exported, but the binary is non-component. By adding a
header into the appropriate place, this error no longer occurs.
---
 src/regexp/arm64/regexp-macro-assembler-arm64.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/regexp/arm64/regexp-macro-assembler-arm64.h b/src/regexp/arm64/regexp-macro-assembler-arm64.h
index 4c6f579655..0a24056beb 100644
--- a/src/regexp/arm64/regexp-macro-assembler-arm64.h
+++ b/src/regexp/arm64/regexp-macro-assembler-arm64.h
@@ -6,6 +6,7 @@
 #define V8_REGEXP_ARM64_REGEXP_MACRO_ASSEMBLER_ARM64_H_

 #include "src/arm64/assembler-arm64.h"
+#include "src/arm64/assembler-arm64-inl.h"
 #include "src/codegen/macro-assembler.h"
 #include "src/regexp/regexp-macro-assembler.h"

--
2.19.1.windows.1

